server/src/main/java/org/elasticsearch/plugins/RepositoryPlugin.java

```diff
@@ -59,4 +61,13 @@
         return Collections.emptyMap();
     }
 
+    /**
+     * Returns a check that is run on restore. This allows plugins to prevent certain restores from happening.
+     *
+     * returns null if no check is provided
+     */
+    default Consumer<IndexMetadata> addPreRestoreCheck() {
```

Comment:
Just curious why you prefer returning a consumer instead of consuming IndexMetadata?

Reply:
This follows the model of not calling directly into plugins, but plugins providing extensions points (at startup time) that are then being called at runtime.